Terraform v.013 upgrade; PR template; and GH Actions integration

vab opened the pull request:
Developer Checklist

[X ] The README contains the correct list of dependencies, inputs, and outputs (e.g., terraform-docs markdown . has been run)
[N/A] Latest versions of stage.tfvars and prod.tfvars uploaded to mit-tfvars S3 bucket
[N/A] Stakeholder approval has been confirmed (or is not needed)
[N/A] Resources currently deploy to stage workspace
A terraform plan should show Plan: 6 to add, 0 to change, 0 to destroy.

What does this PR do?
This PR updates the Terraform code for tf-mod-alb to Terraform v0.13.

Helpful background context
The specific changes to this code include updating the required Terraform version in versions.tf to 0.13, the MIT Libraries GH actions code was integrated, the MIT Libraries PR template was integrated, and finally some minor documentation and code clean up was done.

I also upgraded the Terraform version in the network build yml files. This may cause build failures until all of the network code is upgraded to v0.13.7. However, doing that upgrade now makes this PR a clean v0.13.7 upgrade. We will not have to come back and make a new PR after the network code upgrade is complete.

Requires Database Migrations?
NO

Includes new or updated dependencies?
NO

vab committed:
TF v0.13 upgrade

vab committed:
Why these changes are being introduced:
We are upgrading all of our Terraform modules to be compatible with
Terraform 0.13.

How this addresses that need:
* versions.tf updated to require Terraform 0.13
* versions.tf updated to include required_provider
* PR template added
* tf-validate GitHub Actions workflow added
* Documentation clean up
* outputs.tf formatting

Side effects of this change:
None

* https://mitlibraries.atlassian.net/browse/IN-236

Changes to be committed:
       new file:   .github/PULL_REQUEST_TEMPLATE.md
       new file:   .github/workflows/tf-network-apply.yml
       new file:   .github/workflows/tf-network-destroy.yml
       new file:   .github/workflows/tf-plan-dev.yml
       new file:   .github/workflows/tf-plan-global.yml
       new file:   .github/workflows/tf-validate.yml
       deleted:    .travis.yml
       deleted:    Makefile
       modified:   README.md
       modified:   main.tf
       modified:   outputs.tf
       modified:   versions.tf

zotoMIT commented:
Hey Alex, 
This looks pretty good, but I think the actions you decided on here are overkill.  I think a validate is sufficient.  See this one as an example - https://github.com/MITLibraries/tf-mod-latest-ami/blob/master/.github/workflows/tf-validate.yml

If you were going to use these network apply/destroy actions, you'd need to add secrets to this repository so that they could run correctly and make a lot of changes to the code itself(these refer to shared/core for instance).

vab committed:
Why these changes are being introduced:

We are upgrading all of our Terraform modules to be compatible with
Terraform 0.13.

How this addresses that need:
* versions.tf updated to require Terraform 0.13
* versions.tf updated to include required_provider
* PR template added
* tf-validate GitHub Actions workflow added
* Documentation clean up
* outputs.tf formatting

Side effects of this change:
None

* https://mitlibraries.atlassian.net/browse/IN-236

Changes to be committed:
       new file:   .github/PULL_REQUEST_TEMPLATE.md
       new file:   .github/workflows/tf-validate.yml
       deleted:    .travis.yml
       deleted:    Makefile
       modified:   README.md
       modified:   main.tf
       modified:   outputs.tf
       modified:   versions.tf

vab commented:
@zotoMIT I've made the changes that you suggested. (commit: eeb5c5f)

vab committed:
Why these changes are being introduced:
We are upgrading all of our Terraform modules to be compatible with
Terraform 0.13.

How this addresses that need:
* versions.tf updated to require Terraform 0.13
* versions.tf updated to include required_provider
* PR template added
* tf-validate GitHub Actions workflow added
* Documentation clean up
* outputs.tf formatting

Side effects of this change:
None

* https://mitlibraries.atlassian.net/browse/IN-236

Changes to be committed:
       new file:   .github/PULL_REQUEST_TEMPLATE.md
       new file:   .github/workflows/tf-validate.yml
       deleted:    .travis.yml
       deleted:    Makefile
       modified:   README.md
       modified:   main.tf
       modified:   outputs.tf
       modified:   versions.tf

zotoMIT reviewed (approved):
Looks good to me!

tim-rix reviewed (approved):
Looks good to me.
